tensorboard/data/proto/data_provider.proto

```diff
@@ -0,0 +1,295 @@
+}
+
+message ListRunsResponse {
+  repeated Run runs = 1;
+}
+
+message Run {
+  // Unique opaque string identifier.
+  string id = 1;
+  // User-facing name.
+  string name = 2;
+  // Wall time of earliest recorded event, as floating-point seconds since
+  // epoch (same as event file format).
+  double start_time = 3;
+}
+
+message ListScalarsRequest {
+  // ID of experiment in which to query data.
+  string experiment_id = 1;
+  // Required filter for plugin name. If omitted, an empty message is implied.
+  PluginFilter plugin_filter = 2;
+  // Optional filter for time series. If omitted, all time series match.
+  RunTagFilter run_tag_filter = 3;
+}
+
+message ListScalarsResponse {
+  repeated RunEntry runs = 1;
+  message RunEntry {
+    string run_name = 1;
+    repeated TagEntry tags = 2;
+  }
+  message TagEntry {
+    string tag_name = 1;
+    ScalarMetadata time_series = 2;
+  }
+}
+
+message PluginFilter {
+  // Only match data with exactly this plugin name.
+  string plugin_name = 1;
+}
+
+message RunTagFilter {
+  // Optional filter for runs. If omitted, all runs match.
+  RunFilter runs = 1;
+  // Optional filter for tags. If omitted, all tags match.
+  TagFilter tags = 2;
+}
+
+message RunFilter {
+  // Only match runs with exactly one of these names. In particular, if this
+  // list is empty, no runs match.
+  repeated string runs = 1;
+}
+
+message TagFilter {
+  // Only match tags with exactly one of these names. In particular, if this
+  // list is empty, no tags match.
+  repeated string tags = 1;
+}
+
+message ScalarMetadata {
+  // Largest step value of any datum in this time series. Non-negative.
+  int64 max_step = 1;
+  // Largest wall time of any datum in this time series.
+  double max_wall_time = 2;
+  // Atemporal summary metadata for this time series.
+  tensorboard.SummaryMetadata summary_metadata = 3;
+}
+
+message ReadScalarsRequest {
+  // ID of experiment in which to query data.
+  string experiment_id = 1;
+  // Required filter for plugin name. If omitted, an empty message is implied.
+  PluginFilter plugin_filter = 2;
+  // Optional filter for time series. If omitted, all time series match.
+  RunTagFilter run_tag_filter = 3;
+  // Required downsampling specification describing how many points to return
+  // per time series. If omitted, an empty message is implied, which means that
+  // no points will be returned.
+  Downsample downsample = 4;
+}
+
+message Downsample {
+  // Maximum number of points to return. Should be non-negative. Zero means
+  // zero.
+  int64 num_points = 1;
+}
+
+message ReadScalarsResponse {
+  repeated RunEntry runs = 1;
+  message RunEntry {
+    string run_name = 1;
+    repeated TagEntry tags = 2;
+  }
+  message TagEntry {
+    string tag_name = 1;
+    ScalarData data = 2;
+  }
+}
+
+// A column-major sequence of scalar points. Arrays `step`, `wall_time`, and
+// `value` have the same lengths.
+message ScalarData {
+  repeated int64 step = 1 [packed = true];
```

Comment:
In proto3 `packed=true` is on by default, so this isn't really needed: https://developers.google.com/protocol-buffers/docs/encoding#packed

Reply:
I'm moderately opposed to leaving it in, since I feel like it'll create confusion regarding its effect (namely, none, for proto3 files, and all of our proto files in OSS are proto3).  If some repeated varint fields are labeled and some aren't, it could mislead the reader into assuming there's a functional difference or at least some meaningful difference in intent.  Of course we could resolve that by _requiring_ the annotation everywhere, but that seems overzealous given it has no functional effect, and surveying our existing code, the only places that use it are a handful of older TF protos (newer TF protos generally don't seem to use it):

https://cs.opensource.google/search?ss=tensorflow%2Ftensorboard&q=case:yes%20lang:proto%20repeated%5C%20%5Ba-z%5D